src/server.cpp

```diff
@@ -1575,12 +1575,13 @@ void CServer::FreeChannel ( const int iCurChanID )
             --iCurNumChannels;
 
             // move channel IDs down by one starting at the bottom and working up
```

Comment:
> at the bottom

No it doesn't.  It starts at `i` where `vecChannelOrder[i] == iCurChanID` (i.e. the freed channel).

so rather than
```
// move channel IDs down by one starting at the bottom and working up
// also keep the free channels in sorted order by moving lower-numbered ones down
```
have
```
// move channel IDs down by one starting at the freed channel and working up the current channels
// and then free channels until its position in the free list is reached
```

Reply:
OK, yes, to be accurate, you're right. When I originally wrote the comment I was just talking about the direction of scanning (bottom to top when freeing, top to bottom when making space to allocate). I'll update the comment.